Tweak dark theme

Gaurav-Kushwaha-1225 opened the pull request:
## About Changes

This PR updates the color variables in `DesignVariables` and `MessageListTheme` for a slightly higher contrast in dark theme according to the values in the figma file.

## Fixes: #973  

## Screenshots

| Before | After |
| -------------- | ----------------|
| <img src="https://github.com/user-attachments/assets/a538aa01-6791-404a-99d1-91f53700486f">  |  <img src = "https://github.com/user-attachments/assets/a15801b2-90f5-4df6-aa15-309b9b18eecf"> |
| <img src="https://github.com/user-attachments/assets/8ceb2faf-8832-4ae8-b05c-155fa1065884">  |  <img src = "https://github.com/user-attachments/assets/a7912379-93b1-44aa-8087-e054d63e4a81"> |
| <img src="https://github.com/user-attachments/assets/3d169e9c-fd8f-4a83-9c9d-cc98f11653dd">  |  <img src = "https://github.com/user-attachments/assets/a31be1a6-499d-4896-9801-7f3c3239db65"> |

alya commented:
Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

Gaurav-Kushwaha-1225 commented:
> Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

I’ve cleaned up the commit history by removing unnecessary merge commits into the `TweakDarkTheme` branch and combining related changes into a single commit for clarity.

This is my first time using `rebase -i`. During the process, I used `pick`, `squash`, and `drop` commands for various commits. I initially thought there would be separate commit messages for each, but later realized they were combined into one during the rebase. 
Although, it asked to enter three commit messages after rebasing, Idk why it happened. Could you please explain why all commit messages were merged into one? 
I’d like to understand this better for future contributions.

Apologies for any confusion.

Please review the updated PR.

gnprice commented:
The `#git help` channel on chat.zulip.org is a good place to ask questions about using Git.

Gaurav-Kushwaha-1225 commented:
> Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

Hi @alya, please review the updated PR.

alya commented:
Thanks! I'll let other folks take a look at this one.

PIG208 commented:
Hi! Please read through the commit style guide (also previously linked by Alya) linked in our README and fix the commit messagee: https://github.com/zulip/zulip-flutter?tab=readme-ov-file#submitting-a-pull-request

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -56,13 +56,13 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
   MessageListTheme.dark() :
     this._(
       dateSeparator: Colors.white,
```
There is a change to the `dateSeparator` from opaque to 0.2 opacity:
https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3888-6707&m=dev

Gaurav-Kushwaha-1225 replied:
Okay sure, let's have it to `dateSeparator: Colors.white.withValues(alpha: 0.2);`.
But I think it need to be updated in the design then, since it's opacity is 100% in the design file.
I will update soon.

PIG208 commented on lib/widgets/theme.dart:
```diff
@@ -189,12 +189,12 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
       labelMenuButton: const Color(0xffffffff).withValues(alpha: 0.85),
       mainBackground: const Color(0xff1d1d1d),
       textInput: const Color(0xffffffff).withValues(alpha: 0.9),
-      title: const Color(0xffffffff),
```
Looking at [this](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3863-15744&t=gda9a58taVPMBXlH-0), I see an opportunity to remove `colorMessageHeaderIconInteractive` (below this) and use `title` with the right opacity at places where `colorMessageHeaderIconInteractive` is used. Because it is adjusting something that was not previously a part of the design, let's do that in a separate commit.

Gaurav-Kushwaha-1225 replied:
If `colorMessageHeaderIconInteractive` is removed from `DesignVariables`, it must be removed from both the light and dark themes. 
In the dark theme, `colorMessageHeaderIconInteractive` is identical to `title` but with a different opacity. 
However, in the light theme, `colorMessageHeaderIconInteractive` and `title` have distinct values (`Colors.black.withValues(alpha: 0.2)` and `Color(0xff1a1a1a)`), not just a difference in opacity.

PIG208 replied:
I see. Let's make sure that we stick to the design. In Figma, I see that we use `title` for that element, so we should do just that.

It was also hinted that `designVariables.colorMessageHeaderIconInteractive` is meant to be temporary before we have a color specified in a design:
```
              // TODO(design) copies the recipient header in web; is there a better color?
              color: designVariables.colorMessageHeaderIconInteractive, size: 14)),
```
so it should be fine to remove it.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -56,13 +56,13 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
+      dateSeparatorText: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
+      dmRecipientHeaderBg: const HSLColor.fromAHSL(1, 0, 0, 0.14).toColor(),
+      messageTimestamp: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
+      recipientHeaderText: const HSLColor.fromAHSL(0.9, 0, 0, 1).toColor(),
```
Both [this](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=6262-90970&t=gda9a58taVPMBXlH-0) and [`senderName`](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=6262-90970&t=gda9a58taVPMBXlH-0) are the same as `DesignVariables.title` (and `title` is the variable used in Figma), so let's use that for consistency with the design, and remove these variables, all in its separate commit.

Gaurav-Kushwaha-1225 replied:
Here also,
If `senderName` and `recipientHeaderText` is removed from `MessageListTheme` (and use `title` at places where it is used), it must be removed from both the light and dark themes.

In the dark theme, both these values are identical to title but with a different opacity.
```dart
      recipientHeaderText: const HSLColor.fromAHSL(0.9, 0, 0, 1).toColor(),
      senderName: const HSLColor.fromAHSL(0.9, 0, 0, 1).toColor(),
```
```dart
      title: const Color(0xffffffff),
```
However, in the light theme, both of them and title have distinct values not just a difference in opacity.
```dart
      recipientHeaderText: const HSLColor.fromAHSL(1, 0, 0, 0.15).toColor(),
      senderName: const HSLColor.fromAHSL(1, 0, 0, 0.2).toColor(),
```
```dart
      title: const Color(0xff1a1a1a),
```

PIG208 replied:
Similarly, we should refer to the Figma design. Because the design uses `title` in light/dark mode, we should use the same variable as well.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -56,13 +56,13 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
-      recipientHeaderText: const HSLColor.fromAHSL(0.8, 0, 0, 1).toColor(),
+      dateSeparatorText: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
+      dmRecipientHeaderBg: const HSLColor.fromAHSL(1, 0, 0, 0.14).toColor(),
+      messageTimestamp: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
```
In the new design, `labelTime` is the variable used for both [this](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3888-6647&m=dev) and [dateSeparatorText](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3863-15897&m=dev). Let's introduce that variable replacing these two that we have, in its own commit.

Gaurav-Kushwaha-1225 replied:
> In the new design, `labelTime` is the variable used for both [this](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3888-6647&m=dev) and [dateSeparatorText](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3863-15897&m=dev). Let's introduce that variable replacing these two that we have, in its own commit.

Are you referring this in above comment to `dmRecipientHeaderBg`?

PIG208 replied:
"This" as in the variable on the line I commented on, i.e. `messageTimestamp`.

PIG208 commented on lib/widgets/theme.dart:
```diff
@@ -189,12 +189,12 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
       foreground: const Color(0xffffffff),
```
We also need to adjust the color of the compose box background ([`composeBoxBg`](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3897-8496&m=dev)), compose box top border ([`borderBar`](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3863-15744&t=gda9a58taVPMBXlH-0)).

PIG208 replied:
I just noticed that we have #1207 (PR #1233) for this. Let's just skip changes to the border bar entirely for this PR, but we still need to update background color.

PIG208 replied:
bump; the background color needs to be updated

PIG208 replied:
I just double-checked and it looks like there is no change to `composeBoxBg`. So there is nothing needed to be done here.

Gaurav-Kushwaha-1225 replied:
Sure, the `composeBoxBg` has Hex(0xFF0F0F0F) color in design file which is same as mentioned in code.

PIG208 reviewed (commented):
Thanks for the PR! I noted the differences between this revision and the Figma design after going through `lib/widgets/message_list.dart` and some part of `lib/widgets/compose_box.dart` for colors used in the widgets.

Some of the comments are just about different colors being used, and others request the addition/removal of variables. It should be fine to include color adjustments all in a single commit, but for adding/removing variables, let's try to have separate commits for the ones added, so it is easier to see which variables they replace. Consider using the [NFC](https://github.com/zulip/zulip-mobile/blob/main/docs/glossary.md#nfc) tag if a commit is a pure refactor.

I think there are also changes to the autocomplete window, but that should be addressed in #995,

Gaurav-Kushwaha-1225 commented:
Hi @PIG208,
I have updated the commits as per your request.

- Made separate commits as required
- Removed `colorMessageHeaderIconInteractive`, `senderName` & `recipientHeaderText`
- Added `labelTime` in `MessageListTheme`

Have a review of these changes and tell if anything else is required regarding this issue.

PIG208 commented on lib/widgets/theme.dart:
```diff
@@ -493,5 +493,5 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
 ChannelColorSwatch colorSwatchFor(BuildContext context, Subscription subscription) {
   return DesignVariables.of(context)
-    .channelColorSwatches.forBaseColor(subscription.color);
+      .channelColorSwatches.forBaseColor(subscription.color);
```
nit: indentation

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -55,14 +55,14 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
 
```
```
theme: Adjust foreground & dateSeparator colors

Adjusted the colors for `foreground` in
`DesignVariables` and `dateSeparator` in `MessageListTheme`
as asked.
```

This commit is adjusting more colors than the commit message is suggesting. Some of them is probably irrelevant because we refactored them away later, right? It will be cleaner to make this the final commit in the sequence.

Gaurav-Kushwaha-1225 replied:
Should I edit the commit message too after reordering the commits, like the below one;

```
theme: Adjust foreground & dateSeparator colors

Adjusted the colors for `foreground` in
`DesignVariables` and `dateSeparator` in `MessageListTheme`
as asked. 
Updated `dateSeparator` and `dmRecipientHeaderBg` values acc. to   
their values in figma file.
`foreground`: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908368175
`dateSeparator`: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908328416
```

PIG208 commented on lib/widgets/theme.dart:
```diff
@@ -147,7 +147,6 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
     bgSearchInput: const Color(0xffe3e3e3),
     textMessage: const Color(0xff262626),
     channelColorSwatches: ChannelColorSwatches.light,
-    colorMessageHeaderIconInteractive: Colors.black.withValues(alpha: 0.2),
```
I know that I mentioned that pure refactors can be marked as "[nfc]", but this wouldn't be an "[nfc]" commit because we changed the color in the process of replacing the variable with `title`.

PIG208 commented on lib/widgets/theme.dart:
```diff
@@ -191,7 +191,7 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
     labelMenuButton: const Color(0xffffffff).withValues(alpha: 0.85),
     mainBackground: const Color(0xff1d1d1d),
     textInput: const Color(0xffffffff).withValues(alpha: 0.9),
-    title: const Color(0xffffffff),
```
nit: this commit can't be "[nfc]" because of the appearance change

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -55,14 +55,14 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
-      recipientHeaderText: const HSLColor.fromAHSL(0.8, 0, 0, 1).toColor(),
+      dateSeparator: Colors.white.withValues(alpha: 51),
+      dateSeparatorText: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
+      dmRecipientHeaderBg: const HSLColor.fromAHSL(1, 0, 0, 0.14).toColor(),
```
Could you identify where this color comes from? I looked at https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3702-28206&p=f&t=dBCq6jmgNML8Joly-0 but couldn't find it.

Gaurav-Kushwaha-1225 replied:
- This `dmRecipientHeaderBg` is the light-colored top bar in below screen.
- Though, it is represented using `bg-top-bar` in the figma design file with value `HSLColor.fromAHSL(1, 0, 0, 0.14).toColor()`.

<img src="https://github.com/user-attachments/assets/38973f1a-063a-492e-ad86-88c625f5e09c" width=50%>

PIG208 replied:
Do you have a link the the design? Selecting the frame and then copying the URL should do. For new colors or adjustments the Figma link should also be included in the commit messages.

Gaurav-Kushwaha-1225 replied:
Sure,
https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4993-20811&t=xbe6lt56wtywbnXX-0
- This is design link for `dmRecipientHeaderBg`.
- As I have mentioned above, it is represented using `bg-top-bar` in the figma design file with value `HSLColor.fromAHSL(1, 0, 0, 0.14).toColor()`.

PIG208 replied:
Thanks for the link!

> As I have mentioned above, it is represented using bg-top-bar in the figma design file with value HSLColor.fromAHSL(1, 0, 0, 0.14).toColor().

It looks like the figma design doesn't have the recipient header (it has the app/top bar, though). If we are using bg-top-bar for the recipient headers, it might be a good idea to just use that same variable and remove `dmRecipientHeaderBg`.

However, given that the Figma design doesn't actually show bg-top-bar being used for that specific message list item, I think we should be fine holding off from updating `dmRecipientHeaderBg`, unless such a change is necessary for it to not look odd? Asking for clarification on this in [#mobile-design](https://chat.zulip.org/#narrow/channel/530-mobile-design) should help.

Gaurav-Kushwaha-1225 replied:
- Raised a query in the [#mobile-design](https://chat.zulip.org/#narrow/channel/530-mobile-design/topic/.60dmRecipientHeaderBg.60.20in.20.60MessageListTheme.60) channel.  

- Reverted changes to `dmRecipientHeaderBg`. If there comes any further decisions on this issue on the chat, the PR will be updated accordingly.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -345,7 +345,7 @@ class MessageListAppBarTitle extends StatelessWidget {
             padding: const EdgeInsetsDirectional.only(start: 4),
```
commit message nit: it would be helpful to mention `title` in the commit summary, i.e.:

```
theme: Replace `colorMessageHeaderIconInteractive` with `title`
```

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -1091,7 +1091,7 @@ class StreamMessageRecipientHeader extends StatelessWidget {
               style: recipientHeaderTextStyle(context))),
           const SizedBox(width: 4),
           // TODO(design) copies the recipient header in web; is there a better color?
```
This resolves the TODO comment, so let's just remove that.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -345,7 +345,7 @@ class MessageListAppBarTitle extends StatelessWidget {
             padding: const EdgeInsetsDirectional.only(start: 4),
             child: Icon(icon,
               // TODO(design) copies the recipient header in web; is there a better color?
```
Same here.

PIG208 reviewed (commented):
Thanks for the update! The changes mostly look good to me, except for the first commit that adjusts a bunch of colors. It turned out that "[nfc]" is not applicable for the commits, and that's fine. They are useful to mark commits that are meant to incur no behavioral changes, so the reviewer can focus on the non-nfc commits.


The final commit of the PR needs a line to fix the issue:

```
...
...

Fixes: #973
```

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -32,9 +32,7 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
       dateSeparator: Colors.black,
```
```
theme[nfc]: Remove senderName & recipientHeaderText 
```

commit message nit: similarly, mentioning `title` in the commit summary is helpful

Gaurav-Kushwaha-1225 replied:
I didn't get why you are saying to add `title` in this commit message.
```
theme: Replace `dateSeparatorText` and `messageTimestamp` with `labelTime`
```
This commit message as you have mentioned above is fine for this commit.
Please correct me if I didn't understand properly at any place.

PIG208 replied:
I was referring to commit 4b5405d2d4d03909df11aa2849292343379036da:

```
theme[nfc]: Remove senderName & recipientHeaderText 
```

where we replace occurences of `senderName` and `recipientHeaderText` with `title` instead.

Gaurav-Kushwaha-1225 replied:
Sure, I got it now.
Thank you

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -30,9 +30,8 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
   MessageListTheme.light() :
```
commit message nit: start the summary message with an imperative verb:

```
theme: Added labelTime variable in MessageListTheme 
```
- (with other changes) ->
```
theme: Replace dateSeparatorText and messageTimestamp with labelTime
```

Gaurav-Kushwaha-1225 commented:
Thanks @PIG208 for the review. 
I will look after the requested changes above and update this PR soon.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -29,9 +29,8 @@ import 'theme.dart';
 class MessageListTheme extends ThemeExtension<MessageListTheme> {
```
```
 theme: Replace dateSeparatorText and messageTimestamp with `labelTime`

Removed `dateSeparatorText` & `messageTimestamp` from
`MessageListTheme` and introduced `labelTime` in place of it
as per the design guidelines.
discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908359783

Fixes: #973
```

Because the commit message repeats the summary and the actual diff, we can remove most of it:

```
 theme: Replace dateSeparatorText and messageTimestamp with `labelTime`

discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908359783

Fixes: #973
```

We should remove the "Fixes" line because this is not that final commit that resolves the issue.

Let's also add a link to the Figma design.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -369,8 +369,7 @@ class MessageListAppBarTitle extends StatelessWidget {
           Padding(
```
```
theme: Replace colorMessageHeaderIconInteractive with title

Removed `colorMessageHeaderIconInteractive` from
`DesignVariables` and used `title` in place of it
as per the design guidelines.
discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908343739
```

We can remove most of the commit message as it repeats the title and the diff, similar to other commits.

Let's also add a link to the Figma design.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -31,9 +31,7 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
     dateSeparator: Colors.black,
```
```
theme: Replace senderName & recipientHeaderText with title

Removed `senderName` & `recipientHeaderText` from
`MessageListTheme` and used `title` from `DesignVariables`
as per the design guidelines.
discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908349307

Fixes: #973
```

Because the commit message repeats the title and the actual diff, we can remove most of it.

```
theme: Replace senderName & recipientHeaderText with title

discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908349307

Fixes: #973
```

Similarly, we should link to the Figma design.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -152,9 +146,8 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
-      dateSeparatorText: Color.lerp(dateSeparatorText, other.dateSeparatorText, t)!,
       dmRecipientHeaderBg: Color.lerp(streamMessageBgDefault, other.dmRecipientHeaderBg, t)!,
-      messageTimestamp: Color.lerp(messageTimestamp, other.messageTimestamp, t)!,
+      labelTime: Color.lerp(streamMessageBgDefault, other.labelTime, t)!,
```
Hmm. This doesn't seem right.

Gaurav-Kushwaha-1225 replied:
Yeah,
Sorry for this, it will be updated accordingly soon.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -54,12 +51,9 @@ class MessageListTheme extends ThemeExtension<MessageListTheme> {
+    labelTime: const HSLColor.fromAHSL(0.5, 0, 0, 1).toColor(),
     senderBotIcon: const HSLColor.fromAHSL(1, 180, 0.05, 0.5).toColor(),
-    senderName: const HSLColor.fromAHSL(0.85, 0, 0, 1).toColor(),
     streamMessageBgDefault: const HSLColor.fromAHSL(1, 0, 0, 0.15).toColor(),
```
Ah, I missed this earlier. Here, `streamMessageBgDefault` does not match the color in the Figma design. The variable corresponding to this appears to be `bg-message-regular`. We should rename this variable and update the color.

Gaurav-Kushwaha-1225 replied:
Sure, I will do that in a separate commit.

Gaurav-Kushwaha-1225 replied:
While, working on this comment, I found this line in `message_list.dart` in main branch on zulip-flutter repo;

- line 156, `dmRecipientHeaderBg: Color.lerp(streamMessageBgDefault, other.dmRecipientHeaderBg, t)!,`

- Here, it should be `dmRecipientHeaderBg: Color.lerp(dmRecipientHeaderBg, other.dmRecipientHeaderBg, t)!,`

Am I right ??

PIG208 replied:
Ah, great find. Yes, this is a bug.

Gaurav-Kushwaha-1225 replied:
Sure, I will update the PR soon with all the requested changes.
Thank you.

Gaurav-Kushwaha-1225 replied:
> Ah, great find. Yes, this is a bug.
> 
> See also: [#1305 (comment)](https://github.com/zulip/zulip-flutter/pull/1305#discussion_r1929247551)

Since, this issue is already being addressed in https://github.com/zulip/zulip-flutter/pull/1305#discussion_r1929247551 and being resolved there in some commit.

So, while changing the `streamMessageBgDefault` to `bgMessageRegular` what shall I add to this line.
i.e.
- `dmRecipientHeaderBg: Color.lerp(streamMessageBgDefault, other.dmRecipientHeaderBg, t)!`
- `dmRecipientHeaderBg: Color.lerp(bgMessageRegular, other.dmRecipientHeaderBg, t)!`

PIG208 replied:
It should be fine to include a fix in your PR, because it is fairly trivial.

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -29,12 +29,9 @@ import 'theme.dart';
 class MessageListTheme extends ThemeExtension<MessageListTheme> {
   static final light = MessageListTheme._(
     dateSeparator: Colors.black,
```
I probably have missed this one too, or maybe it was added — `dateSeparator` doesn't exist as a variable in Figma, instead the color of the separators is just `DesignVariables.foreground`. Let's remove this variable and update the references.

PIG208 reviewed (commented):
Thanks for the update! Left some more comments.

Gaurav-Kushwaha-1225 commented:
Hi @PIG208 , 
I have updated this PR according to your review.

Please have a look at it and let me know if anything else is required.
Thank you! 😊

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -369,8 +369,7 @@ class MessageListAppBarTitle extends StatelessWidget {
             child: Icon(icon,
-              // TODO(design) copies the recipient header in web; is there a better color?
-              color: designVariables.colorMessageHeaderIconInteractive, size: 14)),
+              color: designVariables.title, size: 14)),
```
This should be `designVariables.title.withFadedAlpha(0.5)` and the same for the other location.

https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3941-7975&t=dgExngy3z67AN7HZ-0

Gaurav-Kushwaha-1225 replied:
Umm, this gives color to icons beside topic names in the chat section.

In below screenshot, it shows the color of the icon it represents. And, it's a white color with alpha 0.9.

<img src="https://github.com/user-attachments/assets/7b9fc6d9-2115-4ace-a69c-3b8c9880a74c" width=50%>

PIG208 replied:
![image](https://github.com/user-attachments/assets/f06e4c40-abdf-4f1e-894b-892279d506ce)
Hmm yeah this can be confusing. The opacity is set separately here, which is also why we need `withFadedAlpha` because the design variable itself can have a different alpha value.

PIG208 replied:
Yeah, that sounds good to me.

PIG208 reviewed (commented):
Thanks for the update! Left a comment. Could you also post the updated screenshots?

PIG208 commented on lib/widgets/message_list.dart:
```diff
@@ -29,7 +29,6 @@ import 'theme.dart';
 class MessageListTheme extends ThemeExtension<MessageListTheme> {
```
```
theme: Replace dateSeparator with foreground

Figma: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4993-20811&t=vu2wWOBiVb0BB6Vz-0
discussion: #1213 (comment)

Fixes: #1213
```

The "Fixes" part should refer to the issue number.

Gaurav-Kushwaha-1225 commented:
## Screenshots of all Commits

- ### Commit: https://github.com/zulip/zulip-flutter/pull/1213/commits/885f1fd94463e5a65edfe300517c3e8c48e9ffe6, https://github.com/zulip/zulip-flutter/pull/1213/commits/a4996d63aa493c68d9b5ab4f9562e7f91af3e5ba i.e. The Topic row icon color and background color of message.
| Before | After |
| -------- | -------- |
| ![photo_2025-02-04_13-39-21](https://github.com/user-attachments/assets/937b1961-f45a-427d-b5a8-3ea63888acd3) | ![photo_2025-02-04_13-39-23](https://github.com/user-attachments/assets/d0d16670-beff-4372-abe9-6302366b6857) |
| ![photo_2025-02-04_13-39-19](https://github.com/user-attachments/assets/25f6d603-c371-4cdb-b2ab-78fa6d62a440) | ![photo_2025-02-04_13-39-25](https://github.com/user-attachments/assets/ad74135a-bda0-4501-9b47-c4d179417d96) |

- ### Commits: https://github.com/zulip/zulip-flutter/pull/1213/commits/c94e0e2fb5f6e4af2d30469a9557452ba00c1454, https://github.com/zulip/zulip-flutter/pull/1213/commits/752648feb192965f608fa73e3aad246fa7e90a0d, https://github.com/zulip/zulip-flutter/pull/1213/commits/06a5138faf52d77f0f3d36934b2b898f12eadd1c

| Before | After |
| -------- | -------- |
| ![photo_2025-02-04_13-39-15](https://github.com/user-attachments/assets/a99b499a-186d-4086-acca-89e9d2aff3bd) | ![photo_2025-02-04_13-39-29](https://github.com/user-attachments/assets/61e6fac7-b44a-454b-85a2-5a5aeb98c8b4) |
| ![photo_2025-02-04_13-39-18](https://github.com/user-attachments/assets/17f777fd-f375-4f63-8643-281256a89f94) | ![photo_2025-02-04_13-39-27](https://github.com/user-attachments/assets/e4fd18c9-8265-4488-9b65-667b703d3524) |

alya commented:
What's going on with the dramatic changes at the bottoms of your screnshots?

PIG208 commented:
Apart from the issue Alya brought up, the update looks good to me! Marking this for Greg's review.

Gaurav-Kushwaha-1225 commented:
> What's going on with the dramatic changes at the bottoms of your screenshots?

It seems to be a new issue in the main branch. When you change the theme after opening the app (like you are in the DMs page and change the theme from there), the bottom taskbar color doesn’t update properly. 

I’ll check if there’s an existing issue for this or file a new one.

Thanks for pointing it out!

Gaurav-Kushwaha-1225 commented:
> What's going on with the dramatic changes at the bottoms of your screenshots?

Hi @alya, Thanks for pointing out the issue, I have filed an issue for this.
Issue: #1333

Gaurav-Kushwaha-1225 commented:
Hii @all,
There were some merge conflicts, I have resolved them all and it is ready for review now.
Thank You. 😉

gnprice commented:
Thanks for working on this.

For reviewing this PR, the main thing I need to be able to do is to understand what's changing and how that relates to the design recorded in Figma. A key part of the task of writing a PR is to present and explain the changes so as to make that easy. This version doesn't currently do that.

For example, the first commit has this commit message:

    theme: Replace `colorMessageHeaderIconInteractive` with `title`
    
    Figma: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4993-20811&t=C1CYlaJaG8QWOHqG-0
    discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908343739

and it changes two places in the code. Those two places were indeed using `designVariables.colorMessageHeaderIconInteractive` and they get changed to use `designVariables.title`. But what's the actual effect of that in the UI? And what's the reason for that change?

In particular, the Figma link goes to a screen that looks like this:
![image](https://github.com/user-attachments/assets/cdd27e90-25dd-4cc8-bfda-64838d68e483)

As far as I can tell, that screen doesn't contain any examples of either of the two places in the code that get changed in this commit. So if it has any relevance to explaining this commit, I don't see how.

@Gaurav-Kushwaha-1225 please give this PR another round of [self-review](https://zulip.readthedocs.io/en/latest/contributing/reviewable-prs.html#review-your-own-work), with particular attention to explaining your changes, after first rereading our guidelines on commit messages:
https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages
Figma links are very helpful for this sort of change, but make sure they're pointing to the place you intend them to. And in the commit messages, describe what the changes mean for what the user sees in the UI. Mentioning identifiers from the code (as these commit messages focus on doing) can be helpful too, but is less essential because that's something the reader will see immediately upon looking at the diff.

Gaurav-Kushwaha-1225 commented:
Hi @gnprice,

Thank you for the detailed feedback. I sincerely apologize for the lack of clarity in my commit messages.

I initially included a more detailed explanation, but it was revised based on feedback from @PIG208. For reference, my original commit message was:

```
theme: Replace `colorMessageHeaderIconInteractive` with `title`

Removed colorMessageHeaderIconInteractive from DesignVariables and replaced it with title as per the design guidelines.

Discussion: [link]
```

I understand that commit messages should not only describe what changed but also why it changed and how it affects the UI.

Additionally, Will you suggest the sample commit message for this above change according to you as that would be helpful for my future contributions? 
Referring to that I will update the other commit messages too if required?

The change was in the design file the variable used for message header icon was title from `DesignVariables`, so I just removed `colorMessageHeaderIconInteractive` with `title` from `DesignVariables` with `FadedAlpha` as asked in this comment https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1938047329.

gnprice commented:
Yeah, the revision from that original commit message to the current one seems like an improvement — the original doesn't really add any information that isn't in the current revision.

I can't write the commit message for that commit offhand, because I don't know the answers to the questions about what it does. :-) I could find out the answers if I spend time studying the change — but as the author you should already know what your change does, and it's part of your job to explain it.

Here's a bit of a prompt, though, just expanding those questions in a bit more detail. This commit is changing a couple of spots in the code which look like two different spots in the UI. What are those places in the UI? And what's the effect of the change — does it make a particular thing darker in color? Lighter? Some other effect?

Gaurav-Kushwaha-1225 commented:
Hi @gnprice,
I have updated the commit messages as required and below are the commits with short description and screenshots for you to review.

- ### Commit 1: 8632239667fd9da2f9c42fd054885b31f45c1a4e
Description: The `colorMessageHeaderIconInteractive` is replaced with `title` in the `DesignVariables`, since the `title` variable is used at `colorMessageHeaderIconInteractive` places in the Figma Design. See the icon beside 'Latest App Version' topic name.

- ### Commit 2: 5140fc0aa47a10c434a5db95d20bb5c03654d83a
Description: The `dateSeparatorText` and `messageTimestamp` are replaced with `labelTime` in the `MessageListTheme`. Since, `labelTime` is the variable that is used at `dateSeparatorText` and `messageTimestamp` places in the Figma Design.
See the widget that shows date between the divider and time of all messages.

- ### Commit 3: 3565916ddbc40b2f0433d014d560239fb8a7e843
Description: The `senderName` and `recipientHeaderText` in `MessageListTheme` are replaced with `title` in `DesignVariables`, since the `title` variable is used at `senderName` and `recipientHeaderText` places in the Figma Design.
See the name of message sender like 'Gaurav Kushwaha' in below case and the icon beside 'You with yourself' in screenshots.

- ### Commit 4: 680a68c604d828041d924089eceef7b50b4abb90
Description: The `streamMessageBgDefault` variable in `MessageListTheme` is replaced with `bgMessageRegular` variable. Since, this new `bgMessageRegular` variable is being used at `streamMessageBgDefault` places in the Figma Design.
See the background color of messages section especially the color behind the message separating divider.

- ### Commit 5: 89c18a6d676d9760630c13cdbcf34fc5ab2c4a56
Description: The `dateSeparator` variable in `MessageListTheme` is replaced with `foreground` variable of `DesignVariables`, since this `foreground` variable is being used in Figma Design for the same purpose.
See the message separating divider color.


- ## Commits below screenshots are referring to:  8632239667fd9da2f9c42fd054885b31f45c1a4e

| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/937b1961-f45a-427d-b5a8-3ea63888acd3" width=80%> | <img src="https://github.com/user-attachments/assets/d0d16670-beff-4372-abe9-6302366b6857" width=80%> |
| <img src="https://github.com/user-attachments/assets/25f6d603-c371-4cdb-b2ab-78fa6d62a440" width=80%> | <img src="https://github.com/user-attachments/assets/ad74135a-bda0-4501-9b47-c4d179417d96" width=80%> |

- ## Commits below screenshots are referring to: 5140fc0aa47a10c434a5db95d20bb5c03654d83a, 3565916ddbc40b2f0433d014d560239fb8a7e843, 680a68c604d828041d924089eceef7b50b4abb90, 89c18a6d676d9760630c13cdbcf34fc5ab2c4a56

| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/a99b499a-186d-4086-acca-89e9d2aff3bd" width=80%> | <img src="https://github.com/user-attachments/assets/61e6fac7-b44a-454b-85a2-5a5aeb98c8b4" width=80%> |
| <img src="https://github.com/user-attachments/assets/17f777fd-f375-4f63-8643-281256a89f94" width=80%> | <img src="https://github.com/user-attachments/assets/e4fd18c9-8265-4488-9b65-667b703d3524" width=80%> |

Hope this comment resolves the matter.

gnprice commented:
The new commit messages don't contain any added information — they just repeat the same information three times.

Here's the new first commit message:

    theme: Replace `colorMessageHeaderIconInteractive` with `title`
    
    The `colorMessageHeaderIconInteractive` is replaced with `title` in the
    `DesignVariables`, since the `title` variable is used at
    `colorMessageHeaderIconInteractive` places in the Figma Design.
    
    Figma: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4021-7569&t=b56sVJ6in9zEY1tU-0
    discussion: https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1908343739, https://github.com/zulip/zulip-flutter/pull/1213#discussion_r1938047329

This still doesn't answer the questions I asked above:

> Those two places were indeed using `designVariables.colorMessageHeaderIconInteractive` and they get changed to use `designVariables.title`. But what's the actual effect of that in the UI? And what's the reason for that change?

nor the expanded version I offered later:

> Here's a bit of a prompt, though, just expanding those questions in a bit more detail. This commit is changing a couple of spots in the code which look like two different spots in the UI. What are those places in the UI? And what's the effect of the change — does it make a particular thing darker in color? Lighter? Some other effect?

The other commit messages are all the same way.

---

IOW the new commit messages do the same thing as the original commit message you quoted at https://github.com/zulip/zulip-flutter/pull/1213#issuecomment-2650193870 — which @PIG208 correctly told you was redundant, as I told you at https://github.com/zulip/zulip-flutter/pull/1213#issuecomment-2650234761 I agreed with — but they do it even more so.

---

The one other change is that the Figma link in the first commit message got changed (oh and I guess a second link got added under "discussion"). But the screen pointed to by the new link still doesn't look like it has any examples of the UI that is changed in that commit.

Gaurav-Kushwaha-1225 commented:
> This still doesn't answer the questions I asked above:

> > Those two places were indeed using designVariables.colorMessageHeaderIconInteractive and they get changed to use designVariables.title. But what's the actual effect of that in the UI? And what's the reason for that change?

> nor the expanded version I offered later:

> > Here's a bit of a prompt, though, just expanding those questions in a bit more detail. This commit is changing a couple of spots in the code which look like two different spots in the UI. What are those places in the UI? And what's the effect of the change — does it make a particular thing darker in color? Lighter? Some other effect?

Let's take the example of this commit itself.

I have added this below line in above comment and added screenshots to refer I have attached.
> See the icon beside 'Latest App Version' topic name.

or the follow icon in a topic name. Which shows the actual effect on UI? i.e. Makes the icon lighter 

And in the commit message I have added this.
> The `colorMessageHeaderIconInteractive` is replaced with `title` in the `DesignVariables`, since the `title` variable is used at `colorMessageHeaderIconInteractive` places in the Figma Design.

Which explains that in Figma designs this `colorMessageHeaderIconInteractive` variable is not being used to give color to what it is giving to in the app instead `title` variable from `DesignVariables` is being used for that and to keep the code consistent with the flow of design this has to be updated.

And the design frame that it is referring to in the commit link also has a similar kind of drop-down icon beside the topic name.

Gaurav-Kushwaha-1225 commented:
I have added this small explanation above.

If still I am not able to present this correctly, 
I request @PIG208 to please help me regarding this as he only reviewed and guided me through this PR.

And I will update the commit messages once again as told earlier by @PIG208 and @gnprice too.
Thank You.

gnprice commented:
> See the icon beside 'Latest App Version' topic name.

I see, that's in your comment above at https://github.com/zulip/zulip-flutter/pull/1213#issuecomment-2656558904 .

It's not in your actual commit message in the current revision of the PR, though. That commit is here:
https://github.com/zulip/zulip-flutter/commit/8632239667fd9da2f9c42fd054885b31f45c1a4e
If you intended that text to be in your commit message, then that's a good example of why it's important to reread your work before sending it for review:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#reviewing-your-own-code

That has the right shape to begin to answer the questions, then. But I still don't know what it means — the text "Latest App Version" doesn't seem to appear anywhere on the screen that [the commit's (new) Figma link](https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4021-7569&t=b56sVJ6in9zEY1tU-0) points to:
![image](https://github.com/user-attachments/assets/03a03967-4142-4b27-ad39-193430b42a48)


> or the follow icon in a topic name.

This sounds relevant (the information "the follow icon") but also isn't currently in the actual commit message.


> Which shows the actual effect on UI? i.e. Makes the icon lighter

Great, that's an answer to my other main question. It should therefore also go in the commit message :slightly_smiling_face: (and I don't see it in your above comment https://github.com/zulip/zulip-flutter/pull/1213#issuecomment-2656558904 either).

---

I don't think there's any information you need from @PIG208 which you haven't already gotten in this thread. Please try revising the commits again to answer the questions I've asked about the changes and to address the other feedback you've gotten.

If you have other questions, #mobile-dev-help is the best place to ask for help.

Gaurav-Kushwaha-1225 commented:
> I have added this below line in above comment and added screenshots to refer I have attached.

> > See the icon beside 'Latest App Version' topic name.

> or the follow icon in a topic name. Which shows the actual effect on UI? i.e. Makes the icon lighter

I was referring 'Latest App Version' to the screenshots I have attached in my above comment.
https://github.com/zulip/zulip-flutter/pull/1213#issuecomment-2656558904

Anyways, I will ask in #mobile-dev-help regarding commit messages and update this PR soon.

Thank you.

Gaurav-Kushwaha-1225 commented:
Hi @gnprice,
I have updated the commit messages as required and below are the commits with a short description and screenshots for you to review.

- ### Commit 1: 266833ebae5a1b128c3443e1ecb7cccd89d3ba42
| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/6ee0b1d4-735a-440f-a304-8942a24ab7a0" width=80%> | <img src="https://github.com/user-attachments/assets/75db5c42-7375-4db6-8c7c-29e9bd4a200c" width=80%> |
| <img src="https://github.com/user-attachments/assets/00a1befa-867b-4def-857e-c82337626080" width=80%> | <img src="https://github.com/user-attachments/assets/0ec5cbcb-ab70-44a8-bca4-c928ef19b4f0" width=80%> |

- ### Commit 2: 65b5bdbba9afd674479fb21f7f57271bb5c3206a
     - This commit reduces the opacity, causing the time and date stamp to appear lighter/less opaque in both themes. Should I explicitly mention this in the commit message, or would a different phrasing be more appropriate to avoid confusion with "lighter" in other commits?
     
| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/92a072ec-4306-45ad-a46e-e70d27db91e4" width=80%> | <img src="https://github.com/user-attachments/assets/00a10b27-3fd5-4848-86db-f55306146497" width=80%> |
| <img src="https://github.com/user-attachments/assets/2ab68897-90d6-4434-b5c4-d3eca5f3b8a1" width=80%> | <img src="https://github.com/user-attachments/assets/d87e1368-d7a6-4ed0-85e3-fcc60f018fdc" width=80%> |

- ### Commit 3: 4cc0e76a1d1d14963a5ed07ebe18a24d69e4c212
| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/b93751d4-a00e-4938-9f52-c3473af8137e" width=80%> | <img src="https://github.com/user-attachments/assets/1d13dd31-f2c6-4eed-b16f-ed764be40cff" width=80%> |
| <img src="https://github.com/user-attachments/assets/77bf45bb-c281-4679-94f8-4be189839378" width=80%> | <img src="https://github.com/user-attachments/assets/cf4233a6-c401-4027-8ad7-ad93da388b4a" width=80%> |

- ### Commit 4: 91df70784acf3c957f82ebb1630967e779cd9458
| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/35ceff23-55e6-4e8d-9bcb-f5039c430bac" width=80%> | <img src="https://github.com/user-attachments/assets/ad0ccbdb-fcb5-48c9-b5b2-a6e0497c6cd1" width=80%> |
| <img src="https://github.com/user-attachments/assets/4b313058-d349-421d-808d-ea2908404390" width=80%> | <img src="https://github.com/user-attachments/assets/013b959e-91e5-4a20-a3ba-03b32e4ffe34" width=80%> |

- ### Commit 5: 91bd4793f5acc4722a333d123a4397ee5adcc21a
| Before | After |
| -------- | -------- |
| <img src="https://github.com/user-attachments/assets/edb503e2-e54d-429d-aa53-1650250c1f9b" width=80%> | <img src="https://github.com/user-attachments/assets/4b2bc819-2840-4a78-ad37-1c4d1a9d897c" width=80%> |
| <img src="https://github.com/user-attachments/assets/0c581739-bc3c-4f93-b322-a9dcf16f4265" width=80%> | <img src="https://github.com/user-attachments/assets/c43eb95e-2327-495b-a095-569620a56549" width=80%> |

gnprice commented on lib/widgets/message_list.dart:
```diff
@@ -1067,7 +1040,7 @@ class StreamMessageRecipientHeader extends StatelessWidget {
     } else {
       backgroundColor = messageListTheme.unsubscribedStreamRecipientHeaderBg;
-      iconColor = messageListTheme.recipientHeaderText;
+      iconColor = designVariables.title;
```
Note: left icon in the Message Header of Stream message is added just to
    give hspace (see line no. 1075 in `lib/widgets/message_list.dart`).

I don't understand what this remark means.

I think you're referring to this comment in the initializer of streamWidget:
```dart
              child: Icon(size: 16, color: iconColor,
                // A null [Icon.icon] makes a blank space.
                stream != null ? iconDataForStream(stream) : null)),
```

Try looking at the combined feed — you should see the icons there. Or look in the "Mentions" feed — I think you can arrange for that to have some messages in channels you're no longer subscribed to.

gnprice commented on lib/widgets/message_list.dart:
```diff
@@ -928,11 +900,12 @@ class DateSeparator extends StatelessWidget {
+    final designVariables = DesignVariables.of(context);
 
-    final line = BorderSide(width: 0, color: messageListTheme.dateSeparator);
+    final line = BorderSide(width: 0, color: designVariables.foreground);
```
theme: Replace `dateSeparator` with `foreground`
    
    The variable giving color to divider lines in the Message List UI
    i.e. `dateSeparator` has been replaced with the `foreground` variable
    of `DesignVariables`, according to the Figma design.
    There isn't any change in the values of color from the previous one i.e.
    `dateSeparator`.

The fact there's no change in the visible behavior means this commit can be helpfully marked as NFC. See this description:
https://github.com/zulip/zulip-mobile/blob/main/docs/glossary.md#nfc

gnprice replied:
Yeah, the main test for whether something is NFC is: is there any circumstance where a user could see different behavior because of the change?

If there is, then even if it's a subtle effect that only applies in an obscure case, that's a functional change and isn't NFC. But if there isn't, then that's NFC.

(Unless the prefix already says something else that means it's NFC in that sense, for example "test" or "lint". It'd be redundant to label all of those as NFC, so in that case we mark as NFC only if it has no effect on the behavior of the tests or lints etc., in a similar spirit though with different details.)

gnprice reviewed (commented):
Thanks @Gaurav-Kushwaha-1225 for the revision — this is quite a lot better. A few specific comments below.

gnprice commented:
Thanks for the revision! Looks good now; merging.
